File: edu/HRO/CMITDW Taal der Wiskunde/tentamen/Lua/o4.lua

local function round(num)
	return math.floor(num + .5)
end

local function average(t)
	local sum = 0
	for _, v in pairs(t) do
		sum = sum + v
	end
	return sum / #t
end

local function mode(t)
	local v, max = 0, 0
	
	for i = 1, #t do
		local count = 0
		for j = 1, #t do
			if t[i] == t[j] then
				count = count + 1
			end
		end
		if count > max then
			max = count
			v = t[i]
		end
	end
	
	return v
end
	
local function median(t)
	table.sort(t)
	local med = #t / 2
	
	if #t%2 == 0 then
		return (t[med-1] + t[med]) / 2
	else
		return t[med]
	end
end
	
local function variance(t)
	local avg = average(t)
	local sum = 0
	for _, v in ipairs(t) do
		sum = sum + (v-avg)^2
	end
	return sum / #t
end

-- debug methods
local function explode(t)
	for i, v in ipairs(t) do
		print(i, v)
	end
end

local function verifymode(t, a, b)
	for i = a, b do
		local sum = 0
		for _, v in ipairs(t) do
			if (v == i) then
				sum = sum + 1
			end
		end
		print("mode "..i.." = "..sum)
	end
end

local array = {}

for i = 1, 600 do
	array[i] = (1 + (120 + round(100 / math.sin(i))) % 5)
end

do
	--explode(array)
	--verifymode(array, 1, 5)
	
	print("average = "..average(array))
	print("mode = "..mode(array))
	print("median = "..median(array))
	
	local var = variance(array)
	print("variance = "..var)
	print("standard deviation = "..math.sqrt(var))
end

File: edu/HRO/CMITDW Taal der Wiskunde/tentamen/o4.java
import java.util.Arrays;

public class o4
{
	public static void main(String[] args)
	{
		int[] array = new int[600]; // number of ratings
		
		for (int i=0; i<array.length; i++)
			array[i] = 1 + (int) (120 + Math.round(100 * Math.sin(i))) % 5; // 1 to 5
		
		//explode(array);
		//verifymode(array, 1, 5);
		
		System.out.println("average = "+average(array));
		System.out.println("mode = "+mode(array));
		System.out.println("median = "+median(array));
		
		double var = variance(array);
		System.out.println("variance = "+var);
		System.out.println("standard deviation = "+Math.sqrt(var));
	}
	
	public static double average(int t[])
	{
		double sum = 0;
		for (int v : t)
			sum += v;
		return sum / t.length;
	}
	
	public static int mode(int t[])
	{
		int v = 0, max = 0;
		
		for (int i=0; i<t.length; i++)
		{
			int count = 0;
			for (int j=0; j<t.length; j++)
			{
				if (t[i] == t[j])
					count++;
			}
			
			if (count > max)
			{
				max = count;
				v = t[i];
			}
		}
		
		return v;
	}
	
	public static double median(int t[])
	{
		Arrays.sort(t);
		int med = t.length / 2;
		
		if (t.length%2 == 0)
			return (double) (t[med-1] + t[med]) / 2;
		else
			return t[med];
	}
	
	public static double variance(int t[])
	{
		double avg = average(t);
		double sum = 0;
		for (int v : t)
			sum += Math.pow(v-avg, 2);
		return sum / t.length;
	}
	
	// debug methods
	public static void explode(int t[])
	{
		for (int i=0; i<t.length; i++)
			System.out.println(i+"\t"+t[i]);
	}
	
	public static void verifymode(int t[], int a, int b)
	{
		for (int i=a; i<=b; i++)
		{
			int sum = 0;
			for (int v : t)
			{
				if (v == i)
					sum++;
			}
			System.out.println("mode "+i+" = "+sum);
		}
	}
}

